.github/workflows/ci.yml

```diff
@@ -0,0 +1,151 @@
+          - numfocus_nightly: false
+            os: "ubuntu-latest"
+            pyarrow: "0.14.1"
+            python: "3.8"
+          - numfocus_nightly: false
+            os: "ubuntu-latest"
+            pyarrow: "0.15.0"
+            python: "3.8"
+
+    runs-on: ${{ matrix.os }}
+
+    defaults:
+      run:
+        # see https://github.com/goanpeca/setup-miniconda/#important
+        shell: bash -l {0}
+
+    env:
+      IS_MASTER_BUILD: ${{ !matrix.numfocus_nightly && matrix.os == 'ubuntu-latest' && matrix.pyarrow == '0.17.1' && matrix.python == '3.8' }}
+
+    steps:
+      # CI setup
+      - name: Is Master Build?
+        run: echo $IS_MASTER_BUILD
+      - name: Checkout
+        uses: actions/checkout@v2
+      - name: Conda Bootstrap
+        uses: goanpeca/setup-miniconda@v1
+        with:
+          auto-update-conda: true
+          channel-priority: strict
+          channels: conda-forge
+          python-version: ${{ matrix.python }}
+      - name: Conda Config
+        run: |
+          conda config --set always_yes yes
+          conda config --set changeps1 no
+          conda config --set pip_interop_enabled True
+
+      # Installation into the `test` environment
+      - name: Conda Install Pyarrow (non-nightly)
+        run: conda install pyarrow==${{ matrix.pyarrow }}
+        if: matrix.pyarrow != 'nightly'
+      - name: Conda Install Pyarrow (nightly)
+        # Install both arrow-cpp and pyarrow to make sure that we have the
+        # latest nightly of both packages. It is sadly not guaranteed that the
+        # nightlies and the latest release would otherwise work together.
+        run: conda install -c arrow-nightlies arrow-cpp pyarrow
+        if: matrix.pyarrow == 'nightly'
+      - name: Conda Install conda-requirements.txt + conda-test-requirements.txt
+        # run this AFTER pyarrow, since this is less restrictive (so conda doesn't spend too much time downgrading)
+        run: conda install --file=conda-requirements.txt --file=conda-test-requirements.txt
+      - name: Pip Instal NumFOCUS nightly
+        # NumFOCUS nightly wheels, contains numpy and pandas
+        run: env PRE_WHEELS="https://pypi.anaconda.org/scipy-wheels-nightly/simple" pip install --pre --upgrade --timeout=60 -f $PRE_WHEELS pandas numpy
+        if: matrix.numfocus_nightly
+      - name: Pip Install blacken-docs
+        # currently not on conda
+        run: pip install --no-deps blacken-docs
+      - name: Pip Install kartothek
+        run: pip install --no-deps -e .
+      - name: Conda Export
+        run: conda list --export
+
+      # Linters
+      - name: Flake8
+        run: flake8
+      - name: Mypy
+        run: mypy .
+      - name: Black
+        run: black --check .
+      - name: blacken-docs
+        run: blacken-docs
+      - name: Isort
+        run: isort --recursive --check-only
+
+      # Tests
+      - name: Pytest
+        run: pytest --cov kartothek --cov-report xml
+      - name: ASV
```

Comment:
This is new. I'm actually not a big fan of this since it adds quite significant compute time to every job and I wouldn't consider these results very reliable anyhow.
If you want to let these run, I'd prefer to set up a dedicated job which just executes benchmarks but we shouldn't do this for every job in the build matrix

Reply:
> I'm actually not a big fan of this since it adds quite significant compute time to every job and I wouldn't consider these results very reliable anyhow.

They are not reliable (see `--quick`), but we need to check that they're not broken / runnable and IMHO this should be done for every PR. The quick mode should normally be quite fast, the issue with our benchmarks in particular is that they're slow and I think we should fix that (in a later PR, not here).

Reply:
I've changed that to master-job-only (see `IS_MASTER_JOB`). That's way simpler than having a whole new job type and we should rework the benchmarks later. Let me know if that is OK for you.

Reply:
I don't mind executing the for every PR, I agree on this. I also know they are slow and not very well written. Can you open a ticket for this?

I would prefer a dedicate job type but I do understand that this is not easily set up. I believe that's a fair compromise.
IF we wanted to invest in more heterogenous jobs, I also see the opportunity to run docs and linting in dedicated jobs for better / clearer feedback. that's obviously another PR, of course.

Reply:
+1 for running this for every PR, I'd rather detect significant performance issues early. If this is not appropriate now, then we should have it in our backlog

Reply:
@lr4d you won't detect performance issues with this. these runs are only to verify that the benchmarks are actually able to run. Setting it up that we can have an actual reporting is much more complex